feat: Add open_dashboard MCP tool for on-demand dashboard opening

ximot opened the pull request:
## Summary

This PR adds a new MCP tool `open_dashboard` that allows users to open the Serena web dashboard on demand through AI assistance.

**Use case:** Some users prefer to disable automatic browser opening (`web_dashboard_open_on_launch: False`) to avoid focus stealing when Serena starts, but still want easy access to the dashboard when needed. With this tool, they can simply ask the AI to open the dashboard.

## Changes

- **New tool:** `OpenDashboardTool` in `config_tools.py`
- **Agent methods:** `open_dashboard()` and `get_dashboard_url()` in `SerenaAgent`
- **State:** Dashboard URL stored in `_dashboard_url` field
- **Documentation:** Updated `060_dashboard.md` with "Disabling Automatic Browser Opening" section
- **Tool list:** Added `open_dashboard` to `035_tools.md`
- **Config template:** Updated comment to mention the tool as an alternative

## Related Issue

Closes #875 - This provides a solution for users who want dashboard access without automatic browser opening and focus stealing.

## Test Plan

- [x] Format check passes (`uv run poe format`)
- [x] Type check passes (`uv run poe type-check`)
- [x] Python tests pass
- [x] Tested with MCP Inspector - tool works correctly

MischaPanch commented on src/serena/tools/config_tools.py:
```diff
@@ -2,6 +2,19 @@
 from serena.tools import Tool, ToolMarkerDoesNotRequireActiveProject, ToolMarkerOptional
 
 
+class OpenDashboardTool(Tool, ToolMarkerDoesNotRequireActiveProject):
```
should inherit from `ToolMarkerOptional` and documented that the tool should be explicitly enabled through the config. I suspect most users won't want it in their context window

MischaPanch commented on src/serena/resources/serena_config.template.yml:
```diff
@@ -35,7 +35,8 @@ web_dashboard_listen_address: 127.0.0.1
 # whether to open a browser window with the web dashboard when Serena starts (provided that web_dashboard
 # is enabled). If set to False, you can still open the dashboard manually by navigating to
-# http://localhost:24282/dashboard/ in your web browser (24282 = 0x5EDA, SErena DAshboard).
+# http://localhost:24282/dashboard/ in your web browser (24282 = 0x5EDA, SErena DAshboard),
```
same as above (our documentation was off). You'll also have to add that the open_dashboard tool has to be explicitly enabled for the latter version

opcode81 replied:
I made some further updates to the comments.

ximot commented:
> @MischaPanch I updated the PR. Please take a look.
> 
> Note: The original version was not functional - at least not on Windows - because the output redirection required the call to be made from the main thread, but it wasn't when called from the tool. So I now switched from multiprocessing to subprocess.

Thank you for the changes in the code. I will check if there is anything else that needs to be corrected. If there are any comments, I will correct them. 

Today is Sunday, so I needed a bit of rest :)
